code/go/0chain.net/chaincore/chain/entity.go

```diff
@@ -611,6 +618,69 @@ func (c *Chain) setupInitialState(initStates *state.InitStates) util.MerklePatri
 	return pmt
 }
 
+func (c *Chain) addInitialStakes(mpt *util.MerklePatriciaTrie, stakes []state.InitStake) error {
+	var edbDelegatePools []event.DelegatePool
+	for _, v := range stakes {
+		key := v.ProviderType + ":stakepool:" + v.ProviderID
+		sp := stakepool.StakePool{}
+		if err := mpt.GetNodeValue(util.Path(encryption.Hash(key)), &sp); err != nil {
+			if err != util.ErrValueNotPresent {
+				continue
+			}
+			sp.Pools = map[string]*stakepool.DelegatePool{}
+		}
+
+		_, ok := sp.Pools[v.ClientID]
+		if ok {
+			continue
+		}
+
+		sp.Pools[v.ClientID] = &stakepool.DelegatePool{
+			Balance:      v.Tokens,
+			Reward:       0,
+			Status:       0,
+			DelegateID:   v.ClientID,
+			RoundCreated: 0,
+			StakedAt:     common.Timestamp(time.Now().UTC().Unix()),
+		}
+
+		edbDelegatePools = append(edbDelegatePools, event.DelegatePool{
+			PoolID:       v.ClientID,
+			ProviderType: int(spenum.ToProviderID(v.ProviderType)),
+			ProviderID:   v.ProviderID,
+			DelegateID:   v.ClientID,
+			Balance:      v.Tokens,
+			Status:       0,
+			RoundCreated: 0,
+		})
+
+		if _, err := mpt.Insert(util.Path(encryption.Hash(key)), &sp); err != nil {
```

Comment:
Please call  `sp.Save()` to save the stake pool

Reply:
I am trying to use `sp.Save()`, but can't find a suitable value for this parameter `balances cstate.StateContextI`. Apparent at this point in code c.GetStateContextI() is returns nil value and causes panic.

Reply:
See code in line 581, we can pass in the `stateCtx`